frame/staking/src/pallet/impls.rs

```diff
@@ -873,6 +960,37 @@ impl<T: Config> Pallet<T> {
 			DispatchClass::Mandatory,
 		);
 	}
+
+	/// Perform all checks related to both [`Config::TargetList`] and [`Config::VoterList`].
+	#[cfg(any(debug_assertions, feature = "std"))]
+	pub(crate) fn sanity_check_list_providers() {
+		debug_assert_eq!(
+			Nominators::<T>::count() + Validators::<T>::count(),
+			T::VoterList::count()
+		);
+
+		debug_assert_eq!(T::VoterList::sanity_check(), Ok(()));
+		debug_assert_eq!(T::TargetList::sanity_check(), Ok(()));
+		// note that we cannot say much about the count of the target list.
+	}
+
+	#[cfg(any(debug_assertions, feature = "std"))]
+	pub(crate) fn sanity_check_approval_stakes() {
```

Comment:
Just noting that if we do an incremental rollout of the approval stakes, this would be good to check regularly on the live network